enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,172 @@
+custom configuration, but not with all network providers.  With OpenShift-SDN,
+it should be possible to run your own BGP routing daemon on the host via a
+`DaemonSet`.
+
+OVN-Kubernetes uses a different host network configuration where egress
+traffic is routed by Open vSwitch flows programmed by OVN.  It means that a
+dynamic routing daemon must also feed those routes into OVN configuration, so
+it's not as straight forward as just running an existing daemon on the host.
+
+A future enhancement is needed to describe how we would provide this
+functionality natively with OpenShift.
+
+### Pod Network Control Plane
+
+One function that must be implemented for the Pod network is the control plane
+which keeps track of the physical location of each Pod IP address.  BGP is one
+technology which can be used to distribute this information among Nodes in a
+cluster.
+
+There are other ways this control plane functionality can be implemented, such
+as using the Kubernetes API (OpenShift-SDN) or by using some other custom
+control plane technology (OVN-Kubernetes).
+
+Using BGP for this function is *not* a prerequisite for using BGP to
+satisfy other use cases.
+
+### External Service Load Balancing
+
+Kubernetes Services with `type=LoadBalancer` are typically used as an
+abstraction in front of a cloud's load balancer service.  Many on premise
+cluster environments do not have a load balancer service available, so these
+Services may not be used.
+
+One way to expose a Service is to set the `Service.spec.externalIPs` field.
+Once this field has been set, if the IP address(es) are routed to one or more
+Nodes in the cluster, traffic to the Service port(s) will be forwarded to the
+appropriate Service Endpoints.  The major catch with using this interface is
+that routing of these external IP addresses to the Nodes is left up as an
+exercise to the cluster administrator.
+
+Another way to implement `LoadBalancer` Services for these on premise
+clusters is to use BGP speaker on Nodes to publish routes to Nodes where
+`LoadBalancer` IP addresses may be reached.  [MetalLB][2] is one project that
+implements this technique.  [OpenShift Enhancement #356][3] discusses the use
+of MetalLB with OpenShift in more detail.
+
+### Exposing Pods or Services Directly
+
+Some users have expressed an interest in publishing routes for the Pod and
+Service networks via BGP.
+
+If we wanted to offer this, it would not be very difficult.  At least with
+OpenShift-SDN and OVN-Kubernetes, Nodes are already set up to handle traffic
+destined for Pod or Service cluster IP addresses if traffic were to arrive
+destined for those addresses.
+
+To enable this using BGP, a BGP speaker must run on each Node where we would
+like to serve as a router into the cluster network.  Each Node can publish a
+route for the Pod and Service network and upstream routers may use ECMP to
+distribute traffic among the Nodes.
```

Comment:
I wonder if it might be worth explicitly calling out the use case of advertising the same IP in multiple clusters or edge areas (IP Anycast)? It seems to me that this is particularly applicable to edge use cases where a client would be routed to the instance of a particular service IP with the fewest hops possible. This allows for regional affinity as the client is routed to the logically closest Node. This also allows for horizontal scalability, as additional Nodes could be created to spread client load across multiple service instances. Mobile clients may be expected to roam between endpoints as the number of hops changes with location, so this is suited to either fully stateless services over UDP such as DNS or video streaming, or semi-stateless services over TCP such as RESTful API endpoints. This technique could be used to load balance pods or services directly, or to expose multiple LoadBalancer instances.

Reply:
Thank you for the suggested addition!  I've incorporated this text in a new section.